### ISSUE TITLE
When remote notification received, show data and not message

### ISSUE THREAD

**Issue by CGReinhold** [zo0r/react-native-push-notification#675]
So here is my code to the onNotification event:

```javascript
onNotification: notification => {
    console.log('NOTIFICATION:', notification);
    if (AppState.currentState !== 'active') { 
      const { name, message, totalUnread } = notification.data;
      PushNotification.localNotification({
        /* iOS and Android properties */
        title: name, // (optional)
        message, // (required)
      });
      PushNotification.setApplicationIconBadgeNumber(parseInt(totalUnread, 10));
    }
  },
```

and here is the data received in the notification: 

```
{ 
foreground: false,
finish: [Function: finish],
soundName: null,
'google.sent_time': 1521059404126,
'google.ttl': 2419200,
userInteraction: false,
id: '-1487456008',
data: { name: 'Cleyson Reinhold', message: 'abc', totalUnread: '2' },
color: null,
title: null,
'google.message_id': '0:1521059404145846%2a8109ccf9fd7ecd',
message: 'Notification received' 
}
```

My problem is, the notification is always showed with the message "Notification received", and then my local notification using the data. 
Is there a way to not show the notification with the message "Notification received"? Or is my received data wrong formated?

**Comment by CGReinhold** [comment]
It was my bad, the problem was on the data I was sending from server.

**Comment by filipemerker** [comment]
@CGReinhold Would you mind to explain a little better what the problem was? I'm going through the same issue! Thanks!

**Comment by CGReinhold** [comment]
@filipemerker Firstly I had `PushNotification.localNotification` inside of `onNotification`. This means the notification has already showed, so I shouldn't do it again inside of it. That way I removed `PushNotification.localNotification` and got only the 'Notification received' push notification. To solve that one I had to change my server call.
From the server I was sending something like the following:

`{ data: {"title" : "John", "message" : "test","userId" : "1" }}`

Where it should be:

`{{"title" : "John", "message" : "test","userId" : "1" }}`

This way the message attribute received in the notification is filled with my message.

I hope that answers your question.
